# HG changeset patch
# User Max Brister <max@2bass.com>
# Date 1338757222 18000
#      Sun Jun 03 16:00:22 2012 -0500
# Node ID c6728579149c68ab6a05ddd0c9e0264566fc708d
# Parent  1f914446157d32d260828df50abb72f7f7cd6562
Use CXX and CPP flags correctly when configuring llvm

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -736,47 +736,50 @@ LLVM_CPPFLAGS=
 LLVM_LDFLAGS=
 LLVM_LIBS=
 
 dnl llvm-config is messed up
 if  test -n "$LLVM_CONFIG"; then
   LLVM_LDFLAGS="-L`$LLVM_CONFIG --libdir`"
   LLVM_LIBS=`$LLVM_CONFIG --libs`
   dnl Use -isystem so we don't get warnings from llvm headers
-  LLVM_CPPFLAGS="-isystem `$LLVM_CONFIG --includedir`"
+  LLVM_CXXFLAGS="-isystem `$LLVM_CONFIG --includedir`"
 fi
 
 save_CPPFLAGS="$CPPFLAGS"
+save_CXXFLAGS="$CXXFLAGS"
 save_LIBS="$LIBS"
 save_LDFLAGS="$LDFLAGS"
 dnl
 dnl We define some extra flags that llvm requires in order to include headers.
 dnl Idealy we should get these from llvm-config, but llvm-config isn't very
 dnl helpful.
 dnl
-CPPFLAGS="-D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS $LLVM_CPPFLAGS $CPPFLAGS"
+CPPFLAGS="-D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS $CPPFLAGS"
+CXXFLAGS="$LLVM_CXXFLAGS $CXXFLAGS"
 LIBS="$LLVM_LIBS $LIBS"
 LDFLAGS="$LLVM_LDFLAGS $LDFLAGS"
 
 AC_LANG_PUSH(C++)
   AC_CHECK_HEADER([llvm/LLVMContext.h], [
     AC_MSG_CHECKING([for llvm::getGlobalContext in llvm/LLVMContext.h])
       AC_COMPILE_IFELSE(
         [AC_LANG_PROGRAM([[#include <llvm/LLVMContext.h>]],
                          [[llvm::LLVMContext& ctx = llvm::getGlobalContext ();]])],
         [
          AC_MSG_RESULT([yes])
          warn_llvm=
-         XTRA_CXXFLAGS="$XTRA_CXXFLAGS $LLVM_CPPFLAGS"
+         XTRA_CXXFLAGS="$XTRA_CXXFLAGS $LLVM_CXXFLAGS"
         ],
         [AC_MSG_RESULT([no])
     ])
   ])
 AC_LANG_POP(C++)
 CPPFLAGS="$save_CPPFLAGS"
+CXXFLAGS="$save_CXXFLAGS"
 LIBS="$save_LIBS"
 LDFLAGS="$save_LDFLAGS"
 
 if test -z "$warn_llvm"; then
   AC_DEFINE(HAVE_LLVM, 1, [Define if LLVM is available])
 else
   LLVM_CPPFLAGS=
   LLVM_LDFLAGS=
